src/persist-client/src/internal/trace.rs

```diff
@@ -981,9 +985,7 @@ impl<T: Timestamp + Lattice> Spine<T> {
     /// This is a non-public internal method that can panic if we try and insert
     /// into a layer which already contains two batches (and is still in the
     /// process of merging).
-    fn insert_at(&mut self, batch: Option<SpineBatch<T>>, index: usize) {
-        let Some(batch) = batch else { return };
-
+    fn insert_at(&mut self, batch: SpineBatch<T>, index: usize) {
```

Comment:
This commit to remove the Option from insert_at doesn't feel like an overall win to me, but maybe there's some context I'm missing?

Reply:
The motive is purely to not have to worry about stuff like [this comment](https://github.com/MaterializeInc/materialize/pull/26121#discussion_r1530787645).

In the old world, a `None` might represent either a structurally-empty batch or just not having a batch to rollup to a higher level in the rollup method. Since the first case has gone away, it seemed easier to handle the rollup case in the rollup method itself. (And it felt like a bit of a weird behaviour: as though `Vec::push` took an option and just ignored the `None` case, for example.)

Reply:
I don't feel super passionate about this commit, so if that's not motivating I can skip it and find some other way to address your other comment!

Reply:
Our Vacant vs Single(None) slack convo enlightened me here. Keep it!